### PR TITLE
assertion Encode() and few small cleanups from previous review

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: asserts/asserts.go [hunk 1]
```diff
@@ -260,3 +260,14 @@ type assertionTypeRegistration struct {
 }
 
 var typeRegistry = make(map[AssertionType]*assertionTypeRegistration)
+
+// Encode serializes an assertion.
+func Encode(assert Assertion) []byte {
+	content, signature := assert.Signature()
+	needed := len(content) + 2 + len(signature)
```
Comment (zyga) [inline asserts/asserts.go]: What is the 2 for?

File: asserts/asserts.go [hunk 2]
```diff
@@ -260,3 +260,14 @@ type assertionTypeRegistration struct {
...
+	buf := bytes.NewBuffer(make([]byte, 0, needed))
+	buf.Write(content)
+	buf.WriteString("\n\n")
```
Comment (zyga) [inline asserts/asserts.go]: Ah, sorry

File: asserts/asserts_test.go [hunk 3]
```diff
@@ -158,3 +155,21 @@ func (as *AssertsSuite) TestDecodeInvalid(c *C) {
 	}
 
 }
+
+func (as *AssertsSuite) TestEncode(c *C) {
+	encoded := []byte("type: test-only\n" +
```
Comment (zyga) [inline asserts/asserts_test.go]: expected

File: asserts/asserts_test.go [hunk 4]
```diff
@@ -158,3 +155,21 @@ func (as *AssertsSuite) TestDecodeInvalid(c *C) {
...
+		"revision: 5\n" +
+		"header1: value1\n" +
+		"header2: value2\n" +
+		"body-length: 8\n\n" +
+		"THE-BODY" +
+		"\n\n" +
+		"openpgp c2ln")
+	a, err := asserts.Decode(encoded)
+	c.Assert(err, IsNil)
+	encoded2 := asserts.Encode(a)
```
Comment (zyga) [inline asserts/asserts_test.go]: observed or actual

**Comment by mvo5** [comment]
Looks good, +1 (fwiw I agree with zyga about the suggestion "expected" var name in the test, but thats no blocker of course).

File: asserts/asserts.go (hunk 1 above)
Comment (chipaca) [inline asserts/asserts.go]: the 2 is for len(": ")

**Comment by chipaca** [comment]
:+1:

**Pull request merged**